File: typescript/numbers/src/numbers_recursive.ts
export function fibonacci(n: number): number {
  if (n <= 1) return n;
  return fibonacci(n - 1) + fibonacci(n - 2);
}

export function factorial(n: number): number {
  if (n <= 1) return 1;
  return n * factorial(n - 1);
}

export function sum_numbers(n: number): number {
  if (n === 0) return 0;
  return n + sum_numbers(n - 1);
}

File: ocaml/numbers/lib/numbers_recursive.ml
let rec fibonacci n =
  if n <= 0 then 0
  else if n = 1 then 1
  else fibonacci (n - 1) + fibonacci (n - 2)

let rec factorial n =
  if n <= 1 then 1
  else n * factorial (n - 1)

let rec sum_numbers n =
  if n <= 0 then 0
  else n + sum_numbers (n - 1)